doc/langref.html.in

```diff
@@ -9954,6 +9953,164 @@ const c = @cImport({
+      C translation functionality, so on a technical level they are equivalent. In practice,
+      {#syntax#}@cImport{#endsyntax#} is useful as a way to quickly and easily access numeric constants, typedefs,
+      and record types without needing any extra setup. For more complex translation tasks such as
+      function-like macros and C functions, or if you need to pass {#link|cflags|Using -target and -cflags#} to clang,
+      it is recommended to use <code class="shell">zig translate-c</code> and save the results to a file.
+      Doing so makes it easy to view and edit the translated code, for example to change {#syntax#}[*c]T{#endsyntax#}
+      pointers to {#syntax#}[*]T{#endsyntax#} or {#syntax#}*T{#endsyntax#} pointers for improved type safety
+      when dealing with C types in Zig code.
+      </p>
+      {#header_close#}
+      {#see_also|Targets|C Type Primitives|Pointers|C Pointers|Import from C Header File|@cInclude|@cImport#}
+      {#header_close#}
+      {#header_open|C Translation Caching#}
+      <p>
+        The C translation feature (whether used via <code class="shell">zig translate-c</code> or
+        {#syntax#}@cImport{#endsyntax#}) integrates with the Zig caching system. Subsequent runs with
+        the same source file, target, and cflags will use the cache instead of repeatedly translating
+        the same code.
+      </p>
+      <p>
+        To see where the cached files are stored when compiling code that uses {#syntax#}@cImport{#endsyntax#},
+        use the <code class="shell">--verbose-cimport</code> flag:
+      </p>
+      {#code_begin|exe|verbose#}
+      {#link_libc#}
+      {#code_verbose_cimport#}
+const c = @cImport({
+    @cDefine("_NO_CRT_STDIO_INLINE", "1");
+    @cInclude("stdio.h");
+});
+pub fn main() void {
+    _ = c;
+}
+      {#code_end#}
+      <p>
+        <code class="shell">cimport.h</code> contains the file to translate (constructed from calls to
+        {#syntax#}@cInclude{#endsyntax#}, {#syntax#}@cDefine{#endsyntax#}, and {#syntax#}@cUndef{#endsyntax#}),
+        <code class="shell">cimport.h.d</code> is the list of file dependencies, and
+        <code class="shell">cimport.zig</code> contains the translated output.
+      </p>
+      {#see_also|Import from C Header File|C Translation CLI|@cInclude|@cImport#}
+      {#header_close#}
+      {#header_open|Translation failures#}
+      <p>
+        Some C constructs cannot be translated to Zig - for example, <em>goto</em>,
+        structs with bitfields, and token-pasting macros. Zig employs <em>demotion</em> to allow translation
+        to continue in the face of non-translateable entities.
+      </p>
+      <p>
+        Demotion comes in three varieties - {#link|opaque#}, <em>extern</em>, and
+        {#syntax#}@compileError{#endsyntax#}.
+
+        C types that cannot be translated correctly will be translated as {#syntax#}opaque{}{#endsyntax#}.
```

Comment:
```suggestion
        C structs and unions that cannot be translated correctly will be translated as {#syntax#}opaque{}{#endsyntax#}.
```
Might be good to make a distinction here since other untranslateable types are not converted to opaques.